examples/media_element.rs

```diff
@@ -0,0 +1,25 @@
+use web_audio_api::context::{AsBaseAudioContext, AudioContext};
+use web_audio_api::media::{MediaDecoder, MediaElement};
+use web_audio_api::node::{AudioNode, AudioScheduledSourceNode};
+
+fn main() {
+    env_logger::init();
+
+    // construct the decoder
+    let file = std::fs::File::open("sample.ogg").unwrap();
+    let media = MediaDecoder::try_new(file).unwrap();
+
+    // Wrap in a `MediaElement` so buffering/decoding does not take place on the render thread
+    let element = MediaElement::new(media);
+
+    // register the media element node
+    let context = AudioContext::new(None);
+    let node = context.create_media_element_source(element);
+
+    // play media
+    node.connect(&context.destination());
+    node.start();
+
+    // enjoy listening
+    std::thread::sleep(std::time::Duration::from_secs(4));
+}
```

Comment:
Not related to this PR but, in JS this would be written:

```js
const media = new Audio("sample.ogg"); // this is the JS constructor of the html `<audio>` tag
// or
// const media = new Audio();
// media.src = "sample.ogg"
// -> diff - the decoder is hidden in the MediaElement here
const src = audioContext.createMediaElementSource(media);
src.connect(audioContext.destination);

media.start();
// -> diff - this is the media element which holds the start / stop / seek / loop methods, not the WebAudio node
```

I don't say this should be written exactly the same way in Rust, because `Audio` is basically not part of the WebAudio API, but the `MediaElementSourceNode` is maybe more a problem (at least it should be documented as a divergence according to the spec).

Reply:
That's a valid point.

To be fair I'm struggling a bit with the 'boundaries' of the Web Audio API also in other cases. For #77 I was playing around with microphone input and playback and realized I needed to have something like https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder

There are a few conflicting goals:
- stay as close as possible to the spec (as that is the unique selling point of this project, provide the comfort of not learning a new API when coming from a web background)
- leverage all the power of rust (no nullables, no exceptions, use ownership etc)
- create a small library (it's a lot of work already in its current state), versus:
- include related APIs (https://w3c.github.io/mediacapture-main/ https://w3c.github.io/webrtc-pc/ https://html.spec.whatwg.org/multipage/media.html#the-audio-element )
- or include include a few intefaces acting as building blocks (the `MediaStream` trait)

I will create a separate issue for this. It's probably best to steer away from names such as `MediaStream` and `MediaElement` because as you pointed out these are different from the w3c specs. I do think we should include them in the library because you would need them in many applications anyway

Reply:
My general impression is that the current state is not far from a really satisfying compromise. I will try to summarize more clearly my own impressions, and the deviations I have seen, in your new issue.

Reply:
Issue #84